src/backend/distributed/transaction/transaction_management.c

```diff
@@ -695,9 +697,12 @@ IsMultiStatementTransaction(void)
  * MaybeExecutingUDF returns true if we are possibly executing a function call.
  * We use nested level of executor to check this, so this can return true for
  * CTEs, etc. which also start nested executors.
+ *
+ * If the planner is being called from the executor, then we may also be in
+ * a UDF.
  */
 static bool
 MaybeExecutingUDF(void)
 {
-	return ExecutorLevel > 1;
+	return ExecutorLevel > 1 || (PlannerLevel > 0 && ExecutorLevel > 0);
```

Comment:
```suggestion
	return ExecutorLevel > 1 || (ExecutorLevel == 1  && PlannerLevel > 0);
```

By rephrasing it like suggested it will be slightly easier to read at once without changing the behaviour.

By having `ExecutorLevel` at the end and the check which has a lot of overlap it made it hard to understand we always return true for `ExecutorLevel` 2 and above. By getting the `ExecutorLevel` check in the front and a hard equal on `1` it reads more intuitive of what we are checking.